firefox_desktop/views/desktop_daily_retention.view.lkml

```diff
@@ -0,0 +1,355 @@
+  }
+
+  measure: num_clients_dau_in_period {
+    label: "clients - qualified as DAU during the retention period "
+    type: sum
+    sql:  CASE WHEN {% parameter retention_period_picker %} = 1 THEN ${TABLE}.num_clients_dau_on_day WHEN {% parameter retention_period_picker %} = 7 THEN ${TABLE}.num_clients_dau_active_atleastonce_in_last_7_days WHEN {% parameter retention_period_picker %} = 28 THEN ${TABLE}.num_clients_dau_active_atleastonce_in_last_28_days ELSE 0 END;;
+  }
+
+  measure: retention_rate_measure{
+    label: "retention rate"
+    type: number
+    sql: SAFE_DIVIDE(${num_clients_active_in_period}, ${cohort_measure}) ;;
+    value_format: "0.00%"
+  }
+
+  measure: retention_rate_dau_measure{
+    label: "retained as dau rate"
+    type: number
+    sql: SAFE_DIVIDE(${num_clients_dau_in_period}, ${cohort_measure}) ;;
+    value_format: "0.00%"
+  }
+
+# Retention Dimensions
+
+  parameter: retention_period_picker {
+    description: "Which Retention Period to Look Over"
+    type: unquoted
+    allowed_value: {
+      label: "Daily"
+      value: "1"
+    }
+    allowed_value: {
+      label: "Weekly"
+      value: "7"
+    }
+    allowed_value: {
+      label: "Monthly"
+      value: "28"
+    }
+  }
+
+  dimension: retention_period {
+    label: "Retention Period"
+    type:  number
+    description: "Only Works if Selection is Made in Retention Period Picker"
+    sql: IF(MOD(DATE_DIFF(${TABLE}.submission_date, ${TABLE}.first_seen_date, DAY) + 1, {% parameter retention_period_picker %}) = 0, SAFE_DIVIDE(DATE_DIFF(${TABLE}.submission_date, ${TABLE}.first_seen_date, DAY) + 1, {% parameter retention_period_picker %}), NULL) ;;
+  }
+
+
+# Group Dimensions
+  dimension: attribution_campaign {
+    label: "Campaign"
+    type:  string
+    group_label: "Attribution"
+    sql: ${TABLE}.attribution_campaign ;;
+  }
+
+  dimension: attribution_content {
+    label: "Content"
+    type:  string
+    group_label: "Attribution"
+    sql: ${TABLE}.attribution_content ;;
+  }
+
+  dimension: attribution_dlsource {
+    label: "Download Source"
+    type:  string
+    group_label: "Attribution"
+    sql: ${TABLE}.attribution_dlsource ;;
+  }
+
+  dimension: attribution_experiment {
+    label: "Experiment"
+    type:  string
+    group_label: "Attribution"
+    sql: ${TABLE}.attribution_experiment ;;
+  }
+
+  dimension: attribution_medium {
+    label: "Medium"
+    type:  string
+    group_label: "Attribution"
+    sql: ${TABLE}.attribution_medium ;;
+  }
+
+  dimension: attribution_source {
+    label: "Source"
+    type:  string
+    group_label: "Attribution"
+    sql: ${TABLE}.attribution_source ;;
+  }
+
+  dimension: attribution_ua {
+    label: "User Agent"
+    type:  string
+    group_label: "Attribution"
+    sql: ${TABLE}.attribution_ua ;;
+  }
+
+
+# Dates Setup
```

Comment:
I'm assuming this is a pattern we have somewhere? Can we link to it in a comment?

Reply:
I copied this from the [acquisition funnel explore](https://github.com/mozilla/looker-spoke-default/blob/main/firefox_desktop/views/acquisition_funnel.view.lkml) - this explore serves a similar (but still distinct) purpose, so I wanted to make sure that the treatment of dates aligned with that.